rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -48,6 +48,102 @@
 #include "rmw/validate_namespace.h"
 #include "rmw/validate_node_name.h"
 
+namespace
+{
+
+//==============================================================================
+// A function to take ros topic names and convert them to valid Zenoh keys.
+// In particular, Zenoh keys cannot start or end with a /, so this function
+// will strip them out.
+// The Zenoh key is also prefixed with the ros_domain_id.
+// Performance note: at present, this function allocates a new string and copies
+// the old string into it. If this becomes a performance problem, we could consider
+// modifying the topic_name in place. But this means we need to be much more
+// careful about who owns the string.
+z_owned_keyexpr_t ros_topic_name_to_zenoh_key(
+  const char * const topic_name, size_t domain_id, rcutils_allocator_t * allocator)
+{
+  size_t start_offset = 0;
+  size_t topic_name_len = strlen(topic_name);
+  size_t end_offset = topic_name_len;
+
+  if (topic_name_len > 0) {
+    if (topic_name[0] == '/') {
+      // Strip the leading '/'
+      start_offset = 1;
+    }
+    if (topic_name[end_offset - 1] == '/') {
+      // Strip the trailing '/'
+      end_offset -= 1;
+    }
+  }
+
+  std::stringstream domain_ss;
+  domain_ss << domain_id;
+  char * stripped_topic_name = rcutils_strndup(
+    &topic_name[start_offset], end_offset - start_offset, *allocator);
```

Comment:
I know you didn't add this, but we should probably at least put a TODO comment in here for error checking on the `rcutils_strndup`.

Reply:
I added error checking in [ce6af2b](https://github.com/ros2/rmw_zenoh/pull/86/commits/ce6af2b52d67679705dbbb21b8216f2ae77b5b76).